REPOSITORY_OVERVIEW.md

```diff
@@ -0,0 +1,1730 @@
+- **Configurable Base Class**: Most components inherit from `Configuration::Configurable`
+- **Factory Registration**: Components self-register with factories during static initialization
+- **Validation**: Configuration validation happens before machine initialization
+
+### Data Flow
+
+```
+User Config (YAML) → Tokenizer → Parser → Configuration Handlers →
+  → Object Creation (Factories) → Validation → Machine Initialization
+```
+
+---
+
+## Core Components
+
+### 1. Configuration System (`FluidNC/src/Configuration/`)
+
+**Purpose**: Parse and process YAML configuration files
+
+**Key Files**:
+- `Configurable.h` - Base interface for configurable objects
+- `Parser.cpp/h` - YAML parser implementation
+- `Tokenizer.cpp/h` - Breaks YAML into tokens
+- `GenericFactory.h` - Template-based factory for creating objects
+- `ParserHandler.h` - Handles configuration tree building
+- `Validator.cpp/h` - Validates configuration correctness
+
+**Key Concepts**:
+- Configuration happens through the `group()` method which maps YAML keys to member variables
+- Factories use the `InstanceBuilder` pattern for self-registration
+- No dynamic_cast due to ESP32 limitations
+
+### 2. Machine Components (`FluidNC/src/Machine/`)
+
+**Purpose**: Define machine structure and components
+
+**Key Components**:
+- `MachineConfig.cpp/h` - Main machine configuration class
+- `Axes.cpp/h` - Multi-axis management
+- `Axis.cpp/h` - Individual axis configuration
+- `Motor.cpp/h` - Base motor class
+- `Homing.cpp/h` - Homing cycle implementation
+- `LimitPin.cpp/h` - Limit switch handling
+- `SPIBus.cpp/h`, `I2CBus.cpp/h`, `I2SOBus.cpp/h` - Communication bus abstractions
+
+### 3. Motors (`FluidNC/src/Motors/`)
+
+**Purpose**: Stepper and servo motor drivers
+
+**Supported Motor Types**:
+- `StandardStepper` - Basic step/direction stepper
+- `StepStick` - Common stepper driver boards
+- Trinamic drivers:
+  - `TMC2130Driver` (SPI)
+  - `TMC2208Driver` (UART)
+  - `TMC2209Driver` (UART, most common)
+  - `TMC2160Driver`, `TMC5160Driver` (high current)
+- `RcServo` - RC servo motors
+- `Servo` - Standard servo motors
+- `Solenoid` - Solenoid actuators
+- `Dynamixel2` - Dynamixel smart servos
+- `NullMotor` - Placeholder for unused axes
+
+### 4. Spindles (`FluidNC/src/Spindles/`)
+
+**Purpose**: Control spindles and lasers
+
+**Spindle Types**:
+- `PWMSpindle` - PWM-controlled spindles
+- `LaserSpindle` - Laser control with safety features
+- `RelaySpindle` - Simple on/off relay control
+- `OnOffSpindle` - Basic on/off spindle
+- `VFDSpindle` - Variable Frequency Drive control (Modbus)
+- `BESCSpindle` - Brushless ESC control
+- `DacSpindle` - DAC voltage output
+- `10vSpindle` - 0-10V analog control
+- `HBridgeSpindle` - H-bridge motor control
+- `PlasmaSpindle` - Plasma cutter control
+- `NullSpindle` - Disabled spindle
+
+**VFD Protocols** (`FluidNC/src/Spindles/VFD/`):
+- Huanyang protocol variants
+- Modbus RTU communication
```

Comment:
There are many other VFD protocols and none of them are variants of the Huanyang protocol.  The most important one, the generic "ModbusVFD", is not mentioned.